provider/datagen/src/transform/cldr/units/helpers.rs

```diff
@@ -50,18 +50,17 @@ pub fn convert_scientific_notation_to_fraction(
     let exponent = i64::from_str(exponent)
         .map_err(|_| DataError::custom("the exponent is not a valid number"))?;
 
-    let mut result = base;
-    let generic_ten: GenericFraction<BigUint> =
-        GenericFraction::new(BigUint::from(10u32), BigUint::from(1u32));
-    if exponent > 0 {
-        for _ in 0..exponent as u32 {
-            result = result.mul(generic_ten.clone());
-        }
+    let result = if exponent >= 0 {
+        base.mul(GenericFraction::new(
```

Comment:
Consider using [BigRational](https://docs.rs/num-rational/latest/num_rational/type.BigRational.html#) instead of [BigFraction](https://docs.rs/fraction/latest/fraction/prelude/type.BigFraction.html#). The former is part of the `num_*` crates, and it has more functionality, like a `pow` method.

Reply:
it is a big change, but done